aten/src/ATen/dlpack.h

```diff
@@ -80,7 +101,7 @@ typedef struct {
    * \brief The device index.
    * For vanilla CPU memory, pinned memory, or managed memory, this is set to 0.
    */
-  int device_id;
+  int32_t device_id;
```

Comment:
This looks ABI breaking, how is DLPack dealing with the incompatibility arising from it?

Reply:
In an interesting way? https://github.com/dmlc/dlpack/pull/86 is the relevant PR, it motivates this as a clarification of the existing spec. I added a comment there with what I think is the explanation.